blivet/devicetree.py

```diff
@@ -936,6 +936,12 @@ def _hide_ignored_disks(self):
                 if ignored:
                     self.hide(disk)
 
+    def _protect_cdroms(self):
+        # protect cdrom devices
+        for device in self._devices:
+            if device.type == "cdrom":
+                log.debug("protecting cdrom %s", device.name)
+                device.protected = True
```

Comment:
Shouldn't this be done when the CD-ROMs are being "populated-in"?

Reply:
In the optical device populator run() ? Would it be OK to use the flag then?